src/guards.ts

```diff
@@ -0,0 +1,216 @@
+ *    R.isFunction(data)
+ * @returns true if the passed input is a Function, false otherwise
+ * @example
+ *    R.isFunction(() => {}) //=> true
+ *    R.isFunction('somethingElse') //=> false
+ * @category Guard
+ */
+export function isFunction<T>(data: T): data is Extract<T, Function> {
+  return typeof data === 'function';
+}
+
+/**
+ * A function that checks if the passed parameter is Nil (null or undefined) and narrows it's type accordingly
+ * @param data the variable to check
+ * @signature
+ *    R.isNil(data)
+ * @returns true if the passed input is Nil (null or undefined), false otherwise
+ * @example
+ *    R.isNil(undefined) //=> true
+ *    R.isNil(null) //=> true
+ *    R.isNil('somethingElse') //=> false
+ * @category Guard
+ */
+export function isNil<T>(data: T): data is Extract<T, null | undefined> {
+  return data == null;
+}
+
+/**
+ * A function that checks if the passed parameter is an Error and narrows it's type accordingly
+ * @param data the variable to check
+ * @signature
+ *    R.isError(data)
+ * @returns true if the passed input is an Error, false otherwise
+ * @example
+ *    R.isError(new Error('message')) //=> true
+ *    R.isError('somethingElse') //=> false
+ * @category Guard
+ */
+export function isError<T>(data: T): data is Extract<T, Error> {
+  return data instanceof Error;
+}
+
+/**
+ * A function that checks if the passed parameter is a Date and narrows it's type accordingly
+ * @param data the variable to check
+ * @signature
+ *    R.isDate(data)
+ * @returns true if the passed input is a Date, false otherwise
+ * @example
+ *    R.isDate(new Date()) //=> true
+ *    R.isDate('somethingElse') //=> false
+ * @category Guard
+ */
+export function isDate<T>(data: T): data is Extract<T, Date> {
+  return data instanceof Date;
+}
+
+/**
+ * A function that checks if the passed parameter is truthy and narrows it's type accordingly
+ * @param data the variable to check
+ * @signature
+ *    R.isTruthy(data)
+ * @returns true if the passed input is truthy, false otherwise
+ * @example
+ *    R.isTruthy('somethingElse') //=> true
+ *    R.isTruthy(null) //=> false
+ *    R.isTruthy(undefined) //=> false
+ *    R.isTruthy(false) //=> false
+ *    R.isTruthy(0) //=> false
+ *    R.isTruthy('') //=> false
+ * @category Guard
+ */
+export function isTruthy<T>(
+  data: T
+): data is Exclude<T, null | undefined | false | '' | 0> {
+  return !!data;
+}
+
+/**
+ * A function that takes a guard function as predicate and returns a guard that negates it
+ * @param predicate the guard function to negate
+ * @signature
+ *    R.isNot(R.isTruthy)(data)
+ * @returns function A guard function
+ * @example
+ *    R.isNot(R.isTruthy)(false) //=> true
+ *    R.isNot(R.isTruthy)(true) //=> false
+ * @data_last
+ * @category Guard
+ */
+export function isNot<T, S extends T>(
+  predicate: (data: T) => data is S
+): (data: T) => data is Exclude<T, S>;
+export function isNot<T>(predicate: (data: T) => any): (data: T) => boolean;
```

Comment:
not sure why we'd need this extra overload. The tests pass without it - should I remove it?